### PR TITLE
Fix: Updated 404 search link to pre-fill with search parameters

### PR DESCRIPTION
**Pull request by abdullahranginwala**
As mentioned in https://github.com/jenkins-infra/plugin-site/issues/974 ( issue #974 ). I have made the necessary changes to update the search link with appropriate parameters.

### CODE REVIEW

**Comment by halkeye** [comment]
oh cool thanks. Have you checked to see how escaping will work? like if someone does "jenkins plugin", does the link have "jenkins%20plugin" or "jenkins%2520plugin" (which is %20 encoded).

If not, I'll poke around and check when the preview deployment Is up to make sure no xss and stuff, but first glance it looks good.

**Comment by abdullahranginwala** [comment]
Screenshot 2022-02-18 at 11 02 58 AM

Yup, it goes to "jenkins%20plugin"!

**Comment by halkeye** [comment]
build is failing
```
[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-840)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-841)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-842)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-843)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-844)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-845)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-846)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-847)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-848)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-849)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-850)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-851)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-852)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-853)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-854)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-855)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-856)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-857)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-858)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-859)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-860)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-861)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-862)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-863)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-864)[](https://ci.jenkins.io/blue/organizations/jenkins/Infra%2Fplugin-site/detail/PR-1069/3/pipeline#step-51-log-865)[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]: failed Building static HTML for pages - 8.054s

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]: error "location" is not available during server side rendering.

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]: 

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:    6 | const title = 'Plugin page not found';

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:    7 | const notFoundPage = 'pages/404.jsx';

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]: >  8 | const prefillText = location.href.substring(

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:      |                     ^

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:    9 |     location.href.lastIndexOf('/') + 1, location.href.length

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:   10 | );

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:   11 |

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]: 

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]: 

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:   WebpackError: ReferenceError: location is not defined

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:   

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:   - 404.jsx:8 

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:     @jenkins-cd/jenkins-plugin-site/src/pages/404.jsx:8:21

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:   

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:   - bootstrap:19 

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:     @jenkins-cd/jenkins-plugin-site/webpack/bootstrap:19:1

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:   

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:   - static-entry.js:212 

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:     @jenkins-cd/jenkins-plugin-site/.cache/static-entry.js:212:27

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:   

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:   - http.js:145 

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:     node_modules/axios/lib/adapters/http.js:145:1

[2022-02-18T05:43:50.905Z] ➤ YN0000: [@jenkins-cd/jenkins-plugin-site]:
```

Which makes sense, the location object isn't there for the pre-compiled pages. I think its passed in as a property though - https://www.gatsbyjs.com/docs/location-data-from-props/

I would put the logic in the render function directly as its so dead simple and cheap

File: plugins/plugin-site/src/pages/404.jsx [hunk 1]
```diff
@@ -5,6 +5,9 @@ import {Helmet} from 'react-helmet';
 
 const title = 'Plugin page not found';
 const notFoundPage = 'pages/404.jsx';
+const prefillText = location.href.substring(
+    location.href.lastIndexOf('/') + 1, location.href.length
+);
 
 const NotFound = () => (
```
Comment (halkeye) [inline plugins/plugin-site/src/pages/404.jsx]:
```suggestion
const NotFound = ({ location }) => (
```

Comment (halkeye) [reply plugins/plugin-site/src/pages/404.jsx]: in theory

Comment (abdullahranginwala) [reply plugins/plugin-site/src/pages/404.jsx]: Yeah, makes sense. I completely forgot about SSR. I'll give this a try

**Comment by halkeye** [comment]
Tiny little nit pick. The last commit message. Commit messages should describe the change. You described the why but not the what.

"Use location provided by gastby instead of the window object so it works in SSR mode" or something

Not at all preventing merge just good practice

**Comment by halkeye** [comment]
GitHub UI sucks on the phone so i can't check why things failed and it's almost bed time for me. Good luck

**Comment by abdullahranginwala** [comment]
I'm not sure why the tests are failing. I'm trying to find out details but the site isn't opening

**Comment by halkeye** [comment]
>  I'm trying to find out details but the site isn't opening

You can view the results of the ci.jenkins one, not the ci.infra.jenkins (needs VPN)

> I'm not sure why the tests are failing. 

First it was failing due to location not being provided, but I couldn't get it to actually return a testable instance, then it turned out since it was one of the first tests we wrote, its using the wrong rendering tool.

I've updated the tests, and have one failing one now. Looks like your code will work for /discord (untested) but not /discord/. I would recommend making sure both cases are tested

One solution would be something like this
```
"/foo/bar/".split('/').filter(Boolean).reverse()[0]
```

but its your code, so I'll leave it in your hands how you want to handle it.

just FYI, the location object that gatsby returns (pulled in via the react developers tools)
```
{"pathname":"/discord/","search":"","hash":"","href":"http://localhost:3000/discord/","origin":"http://localhost:3000","protocol":"http:","host":"localhost:3000","hostname":"localhost","port":"3000","state":null,"key":"initial"}
```

**Comment by halkeye** [comment]
@abdullahranginwala sorry this took so long, i get easily distracted. Thank you so much for your contribution.

**Pull request merged**